azure-specialized-workloads/oracle/recommendations.yaml

```diff
@@ -0,0 +1,201 @@
+- description: Implement a regional replication strategy for Oracle to meet your workload requirements
+  aprlGuid: dfeb9c7a-7dae-4751-9625-b23a7160a3e1
+  recommendationTypeId: null
+  recommendationControl: Business Continuity
+  recommendationImpact: High
+  recommendationResourceType: Specialized.Workload/oracle
+  recommendationMetadataState: Active
+  longDescription:
+    Regional replication is a key strategy to ensure business continuity and disaster recovery for your Oracle workloads. Implement a regional replication strategy to meet your workload requirements.
```

Comment:
This appears to be more architectural guidance than a configuration for the Oracle resource that can be configured to improve the resiliency of the Oracle resource deployment. As such, wouldn't this better be addressed through WAF guidance?

Reply:
The recommendation is agreed by relevant teams and as such we would like to keep it

Reply:
Agree with @ehaslett this reads as a generic best practice rather than an actionable reliability recommendation. We also already have WAF recommendations RE:05 that cover the same topic.

Reply:
The [Oracle Maximum Availability Architecture for Oracle Database@Azure](https://docs.oracle.com/en/database/oracle/oracle-database/19/haovw/db-azure1.html#GUID-E8360630-A2B8-4A46-9CBF-56EF0BF8A00F) Gold Architecture shows that ODAA can be configured for either zonal or regional replication.  For regional replication the customer would setup an infrastructure/cluster in both regions then either use [Data Guard](https://docs.oracle.com/cd/E11882_01/server.112/e41134/concepts.htm#i1039416) (active/passive) or [Golden Gate](https://docs.oracle.com/goldengate/c1230/gg-winux/GGCON/introduction-oracle-goldengate.htm) (active/active) to replicate either across regions or zones.  

@ehaslett or @ejhenry please describe why this is viewed as a best practice and not an actionable reliability recommendation.

Reply:
> The [Oracle Maximum Availability Architecture for Oracle Database@Azure](https://docs.oracle.com/en/database/oracle/oracle-database/19/haovw/db-azure1.html#GUID-E8360630-A2B8-4A46-9CBF-56EF0BF8A00F) Gold Architecture shows that ODAA can be configured for either zonal or regional replication. For regional replication the customer would setup an infrastructure/cluster in both regions then either use [Data Guard](https://docs.oracle.com/cd/E11882_01/server.112/e41134/concepts.htm#i1039416) (active/passive) or [Golden Gate](https://docs.oracle.com/goldengate/c1230/gg-winux/GGCON/introduction-oracle-goldengate.htm) (active/active) to replicate either across regions or zones.
> 
> @ehaslett or @ejhenry please describe why this is viewed as a best practice and not an actionable reliability recommendation.

I would ask these questions about the recommendation to determine if it is actionable in terms of APRL guidance and remediation:
- Can an ARG query be formed to uncover Azure resource configuration(s) that does not meet this guidance?
- Can a Azure resource configuration that does not meet this guidance be remediated through an Azure resource configuration change?

Reply:
@ehaslett  The configuration change would need to be made through the OCI portal or related APIs.  There is no way to configure Data Guard or Golden Gate through an Azure resource configuration change.  

@basimolimajeed  Due to the above criteria for APRL, this recommendation will need to be moved to WAF documentation.  We could ask that this control pane capability be moved to Azure, but I don't see that happening in the short term.

Reply:
@ehaslett , there are "manual evaluations" that are part of WARA where items cannot be evaluated by ARG queries.  Could we add these items as "manual evaluations" even though they are evaluated in the Oracle portal?  Note that a typical ODAA deployment will have SSO setup between Azure and OCI, but will not be included in Light House access.  So, we would need to ask the customer to validate this for us.

Reply:
@terrymandin I have to defer to @oZakari and @rodrigosantosms on this

Reply:
Hi @terrymandin, it is okay if you add recommendations that cannot be added validated with ARG. If they are valid recommendations that should be checked during a WARA engagement for Oracle, then I think they should be added in.

However, we don't have a field dedicated for remediation, so please just incorporate a link in the learnMoreLink field.

Reply:
@ehaslett , based on @oZakari [feedback](https://github.com/azure/azure-proactive-resiliency-library-v2/pull/463#discussion_r1823494803) indicating that we can include recommendations that cannot be validated with ARG, could you please review the items below and let us know if we we check them in if a remediation link is added?

Reply:
@terrymandin I would again have to defer to @ejhenry, @rodrigosantosms and @oZakari  about whether or not it is appropriate to evaluate an Oracle configuration (i.e. Oracle portal) vs an Azure configuration, and then also provide remediation guidance for something outside of Azure (i.e. Oracle portal).

Another question that still comes to my mind is whether or not:
Can a Azure resource configuration that does not meet this guidance be remediated through an Azure resource configuration change?

Reply:
@oZakari  could you review this PR.  Some of the recommendations were originally rejected as ARG queries could not be written for them.  Your [feedback](https://github.com/azure/azure-proactive-resiliency-library-v2/pull/463#discussion_r1823494803) indicates that recommendations do not necessarily need ARG queries.  In the [comment above](https://github.com/azure/azure-proactive-resiliency-library-v2/pull/463#discussion_r1845061554) @ehaslett is deferring to yourself, Eric or Zach as to whether or not they can be included.  Would you be willing to re-review them?

If possible, can you also re-open the PR?  It was closed by a bot.